src/wallet/db.cpp

```diff
@@ -387,7 +387,7 @@ bool CDB::Rewrite(const string& strFile, const char* pszSkip)
                         while (fSuccess) {
                             CDataStream ssKey(SER_DISK, CLIENT_VERSION);
                             CDataStream ssValue(SER_DISK, CLIENT_VERSION);
-                            int ret = db.ReadAtCursor(pcursor, ssKey, ssValue, DB_NEXT);
+                            ret = db.ReadAtCursor(pcursor, ssKey, ssValue, DB_NEXT);
```

Comment:
I'm not sure promoting these return values up to the function scope is a good idea. Keeping the scope of variables as small as possible ensures that results won't leak or accidentally influence other things.